overlord/devicestate/devicemgr.go

```diff
@@ -1425,6 +1427,73 @@ func (m *DeviceManager) ensurePostFactoryReset() error {
 	return os.Remove(factoryResetMarker)
 }
 
+func (m *DeviceManager) userHasExpired(user *auth.UserState) bool {
+	// If the user has no expiration date, then Expiration should not
+	// be set, and contain the default value.
+	if user.Expiration.IsZero() {
+		return false
+	}
+
+	return user.Expiration.Before(time.Now())
+}
+
+func (m *DeviceManager) removeUser(st *state.State, user *auth.UserState) error {
+	// Remove the user from the system first, if it fails to remove the system
+	// user, then we should not remove it from auth state, otherwise we'd show
+	// a false representation of whether or not that user be present.
+	if err := osutilDelUser(user.Username, &osutil.DelUserOptions{ExtraUsers: !release.OnClassic}); err != nil {
+		return err
+	}
+
+	// And then we remove it from the auth state
+	_, err := auth.RemoveUserByUsername(st, user.Username)
+	return err
+}
```

Comment:
these don't use state on DeviceManager so a bit unclear why make them methods here, also in the PR I mentioneds we grow a new users.go file which is probably the right place for some of this.

Reply:
Updated this after Ondrej's PR was merged!